.github/workflows/reusable_promote_prerelease.yaml

```diff
@@ -0,0 +1,56 @@
+name: Promote Prerelease to Release
+
+# This workflow checks for the latest successful prerelease, excluding any labeled with "artifacts-pending" or "pre-release-failure".
+# If a suitable prerelease is found, it promotes it to a full release. This action should consequently trigger any workflows
+# that respond to the 'released' event type, ultimately invoking 'reusable_on_release.yaml'. This facilitates an automated
+# continuation of the release process.
+
+#  Usage from the caller workflow:
+#  jobs:
+#    promote_prerelease:
+#      uses: newrelic/coreint-automation/.github/workflows/reusable_promote_prerelease.yaml@v2
+#      secrets:
+#        bot_token: ${{ secrets."github_token_name" }}
+
+on:
+  workflow_call:
+    inputs:
+      block_endpoint:
+        description: This endpoint should respond status 200 for the promotion to continue.
+        required: false
+        type: string
+        default: https://newrelic.github.io/coreint-automation/automatic_release_enable
+    secrets:
+      bot_token:
+        description: "github token"
+        required: true
+
+jobs:
+  check_block_endpoint:
+    name: Check block endpoint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check block endpoint
+        run: |
+          http_status=$(curl -s -o /dev/null -w "%{http_code}" "${{ inputs.block_endpoint }}")
+
+          if [ "$http_status" != 200 ]; then
+            echo "Fail checking release block endpoint: ${{ inputs.block_endpoint }}"
+            exit 1
+          fi
+
+  promote_prerelease:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Promote previous prerelease to release
+        env:
+          GH_TOKEN: "${{ secrets.bot_token }}"
+        run: |
+          PRERELEASE_INFO=$(gh release list --repo ${{ github.repository }} --limit 1 --prerelease | grep -v "artifacts-pending" | grep -v "pre-release-failure")
```

Comment:
I do not have --prerelease 
```
gh release list --prerelease
unknown flag: --prerelease

Usage:  gh release list [flags]

Flags:
      --exclude-drafts         Exclude draft releases
      --exclude-pre-releases   Exclude pre-releases
  -q, --jq expression          Filter JSON output using a jq expression
      --json fields            Output JSON with the specified fields
  -L, --limit int              Maximum number of items to fetch (default 30)
  -O, --order string           Order of releases returned: {asc|desc} (default "desc")
  -t, --template string        Format JSON output using a Go template; see "gh help formatting"

```

Reply:
Yes, me neither 😅  This was me testing and I left the wrong command (problems of this not being tested in the PR). It should go without the `--prerelease` and then check manually if it's a prerelease. I updated it.